plugin/builtin/patchtransformer/PatchTransformer_test.go

```diff
@@ -198,6 +198,99 @@ Patch: "something"
 	})
 }
 
+func TestPatchTransformerNotAllowedMultipleStrategicMergePatches(t *testing.T) {
+	th := kusttest_test.MakeEnhancedHarness(t).
+		PrepBuiltin("PatchTransformer")
+	defer th.Reset()
+
+	th.WriteF(`multiplepatches.yaml`, `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: audit
+  namespace: system
```

Comment:
really nit: Would be even better if the `name` and `namespace` of these 2 patch resources matched that of the target resources in `someDeploymentResources`. 

With the given patches in the test, you might suspect that the implementation ignores the `target` when there are multiple strategic merge patches, and only throws the error because ignoring `target`, each patch does not match exactly 1 existing resource in `someDeploymentResources`.